### ISSUE TITLE
useQuery with {ssr: false} is not fetched on client-side

### ISSUE THREAD

**Issue by hevele-moda** [apollographql/apollo-client#8571]
**Intended outcome:**
Invoking a `useQuery` with `{ssr: false}` skips the query on SSR but fetches it on client-side.

**Actual outcome:**
Invoking a `useQuery` with `{ssr: false}` does not fetch the query on client-side either. It returns `{data: undefined, loading: true}` indefinitely.

**How to reproduce the issue:**
- clone this fork https://github.com/hevele-moda/react-apollo-error-template
- run `npm install && npm run`
- Notice that the page shows 'Loading...' and the query never gets resolved.

**Versions**
```
System:
  OS: Linux 5.4 Ubuntu 20.04.2 LTS (Focal Fossa)
Binaries:
  Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
  Yarn: 1.22.10 - /src/discovery/node_modules/.bin/yarn
  npm: 7.20.3 - /src/discovery/node_modules/.bin/npm
Browsers:
  Chrome: 92.0.4515.107
  Firefox: 90.0
npmPackages:
  @apollo/client: ^3.4.1 => 3.4.1 
  apollo: ^2.33.4 => 2.33.4 
  apollo-link-rest: ^0.8.0-beta.0 => 0.8.0-beta.0 
  apollo-storybook-react: ^0.2.4 => 0.2.4 
```

**Comment by benjamn** [comment]
@hevele-moda Thanks for the reproduction!

This appears to be a regression introduced in `@apollo/client@3.4.0-rc.13`, possibly from commit 43d64356e656c42bd28523394e07c1c0584aa4b2. Does that sound plausible @brainkim?

**Comment by brainkim** [comment]
@hevele-moda Sorry for the regression. Was a misunderstanding of the option during a refactoring. Thanks for hte reproduction! Should be fixed in 3.4.3
